server/src/main/java/org/apache/iotdb/db/query/reader/resourceRelated/NewUnseqResourceMergeReader.java

```diff
@@ -88,16 +88,18 @@ public NewUnseqResourceMergeReader(Path seriesPath, TSDataType dataType,
       if (tsFileResource.isClosed()) {
         // get chunk metadata list of current closed tsfile
         currentChunkMetaDataList = DeviceMetaDataCache.getInstance().get(tsFileResource, seriesPath);
-        List<Modification> pathModifications = context
-            .getPathModifications(tsFileResource.getModFile(), seriesPath.getFullPath());
-        if (!pathModifications.isEmpty()) {
-          QueryUtils.modifyChunkMetaData(currentChunkMetaDataList, pathModifications);
-        }
       } else {
-        // metadata list of already flushed chunk groups
+        // metadata list of already flushed chunks
         currentChunkMetaDataList = tsFileResource.getChunkMetaDataList();
       }
 
+      // get modifications and apply to chunk metadatas
+      List<Modification> pathModifications = context
```

Comment:
need a ut? 
Or, in logical, can the PathModifications file be read before the data file is sealed?

Reply:
no bug, I have added some test and comment